python/lsst/validate/base/metric.py

```diff
@@ -55,6 +142,12 @@ class Metric(JsonSerializationMixin):
     description = None
     """Short description of the metric (`str`)."""
 
+    unit = None
```

Comment:
Heads up that in `Specification`-land I'm distinguishing between the string and Astropy Unit object forms.

- `unit` is an `astropy.units.Unit` instance
- `unit_str` is the string representation of a `Unit`. Maybe you could do this here too?